util/top_recommendations.py

```diff
@@ -0,0 +1,39 @@
+import bisect
+
+
+class TopRecommendations(object):
+
+    def __init__(self, n_recommendations):
+        self.recommendations_values = []
+        self.recommendations_indices = []
+        self.n_recommendations = n_recommendations
+
+    def insert(self, index, value):
+        if self.n_recommendations > len(self.recommendations_values):
+            inserted_index = self._insert_and_return_index(self.recommendations_values, value)
+            self.recommendations_indices = self._insert_at_index(self.recommendations_indices, index, inserted_index)
+            return
+        if len(self.recommendations_values) != 0:
+            if self.recommendations_values[0] < value:
+                self.recommendations_values.pop(0)
+                self.recommendations_indices.pop(0)
+                inserted_index = self._insert_and_return_index(self.recommendations_values, value)
+                self.recommendations_indices = self._insert_at_index(self.recommendations_indices,
+                                                                     index, inserted_index)
+
+    def _insert_and_return_index(self, arr, val):
+        index = bisect.bisect(arr, val)
+        bisect.insort(arr, val)
+        return index
+
+    def _insert_at_index(self, arr, val, index):
+        return arr[0: index] + [val] + arr[index:]
```

Comment:
Okay, it's not an issue here, I think. I will leave #18 open if we wanted to recheck these small optimizations in case we needed them later on, or we wanted to export an optimized version.